lib/formatter/phone_number.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'pry'
+
+module Formatter
+  class PhoneNumber
+    InvalidNumber = 'Error!'
+    
+    class << self
+      def call(number)
+        error_message?(number)
+
+        number = remove_whitespace(number)
```

Comment:
I'm not sure what you mean with your feedback on this one

> i don’t know why i need to set the number variable on lines 13 when i don’t need to do that for error_message or formats_area_code

Can you say more?

Reply:
Ah i think i see what you're saying. 

You could change this to: 

```
error_message?(number)
number_without_whitespace = remove_whitespace(number)
formats_area_code(number_without_whitespace)
```

So when you define `number` as a variable and store the result of `remove_whitespace(number)` you're effectively overwriting the `number variable` that you passed into the `call` method. 

Does that answer?